7814 datagrid design updates

tmcconechy opened the pull request:
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

Implements new datagrid header designs. New designs as per: https://www.figma.com/file/4CtBGz8dwvSa6JsBcdtVZQ/Data-Grid?type=design&node-id=2388-30705&mode=design&t=npya2cBC5ZmD3wcz-0

**Related github/jira issue (required)**:
Fixes https://github.com/infor-design/enterprise/issues/7814

**Steps necessary to review your pull request (required)**:
The core of the datagrid can be seen by examining the following set of examples;

- http://localhost:4000/components/datagrid/example-index.html
- http://localhost:4000/components/datagrid/example-multiselect
- http://localhost:4000/components/datagrid/example-reorder.html
- http://localhost:4000/components/datagrid/example-grouped-headers.html 
- http://localhost:4000/components/datagrid/example-nested-grids.html
- Smoke text others if desired http://localhost:4000/components/datagrid/

**Included in this Pull Request**:
- [x] A note to the change log.

**Questions for Design Review**:
1. Slate 03 felt "boxy" for the dark mode so lightened some of the borders http://localhost:4000/components/datagrid/example-index.html?theme=new&mode=dark&colors=default should we go even lighter or darken these to make it look more "smooth"?
2. Select a line on http://localhost:4000/components/datagrid/example-multiselect?theme=new&mode=dark&colors=default should the checkboxes be azure in dark mode? What do you think of the selected state here? 
3. On http://localhost:4000/components/datagrid/example-index.html try sorting a column. Should the different in sort direction up/down icons be more different? (Applies in all themes)
4. In contrast mode, usually we make things one darker. For example the header could be slate 02? Does it look ok identical to light mode? http://localhost:4000/components/datagrid/example-index.html?theme=new&mode=contrast&colors=default

tmcconechy commented:
@talkaboutdesign and @laurensteiger can you check out this fix and please note some questions numbered above that i have. Feel free to chat in teams if you want more clarification on anything.

laurensteiger commented:
@tmcconechy working on this will get back to you thanks

laurensteiger commented:
> **Explain the _details_ for making this change. What existing problem does the pull request solve?**
> 
> Implements new datagrid header designs. New designs as per: https://www.figma.com/file/4CtBGz8dwvSa6JsBcdtVZQ/Data-Grid?type=design&node-id=2388-30705&mode=design&t=npya2cBC5ZmD3wcz-0
> 
> Also removed the "white background header" example from before
> 
> **Related github/jira issue (required)**: Fixes #7814 Fixes #7522 Fixes #7860
> 
> **Steps necessary to review your pull request (required)**: The core of the datagrid can be seen by examining the following set of examples;
> 
> * http://localhost:4000/components/datagrid/example-index.html
> * http://localhost:4000/components/datagrid/example-multiselect
> * http://localhost:4000/components/datagrid/example-reorder.html
> * http://localhost:4000/components/datagrid/example-grouped-headers.html
> * http://localhost:4000/components/datagrid/example-nested-grids.html
> * Smoke text others if desired http://localhost:4000/components/datagrid/
> 
> **Included in this Pull Request**:
> 
> * [x]  A note to the change log.
> 
> **Questions for Design Review**:
> 
> 1. Slate 03 felt "boxy" for the dark mode so lightened some of the borders http://localhost:4000/components/datagrid/example-index.html?theme=new&mode=dark&colors=default should we go even lighter or darken these to make it look more "smooth"?
> 2. Select a line on http://localhost:4000/components/datagrid/example-multiselect?theme=new&mode=dark&colors=default should the checkboxes be azure in dark mode? What do you think of the selected state here?
> 3. On http://localhost:4000/components/datagrid/example-index.html try sorting a column. Should the different in sort direction up/down icons be more different? (Applies in all themes)
> 4. In contrast mode, usually we make things one darker. For example the header could be slate 02? Does it look ok identical to light mode? http://localhost:4000/components/datagrid/example-index.html?theme=new&mode=contrast&colors=default
> 5. Grouped headers https://7814-datagrid-design-updates-enterprise.demo.design.infor.com/components/datagrid/example-grouped-headers.html should they be treated differently as before? I just made them the same which to me makes sense https://main-enterprise.demo.design.infor.com/components/datagrid/example-grouped-headers.html

@tmcconechy 

Question 1
    1. Changed to outline stroke to slate04, to work better with the slate09 cell bg fill. See what border colors are being used currently, change to slate 4 in dark mode. 

Question2
    1. Checkboxes should not be azure in dark mode, change to slate08.
    2. Selected state should be slate 7
    3. Because selected state is slate7, hyperlinks have to move to azure 2 to get at least AA accessibility. 

Question 3
    1. Leave it as is for now. 

Question 4
    1. Keep same 

Question 5
    1. Looks good to me.
    
Additional Note
Also note, please make the cell bgs in Datagrid slate09, it is currently transparent.
You can see all the changes here https://www.figma.com/file/4CtBGz8dwvSa6JsBcdtVZQ/Data-Grid?type=design&node-id=2388-30705&mode=design

tmcconechy commented:
@laurensteiger after making the dark mode slate 90 background and slate 40. I still think it looks super boxy if thats the best word. Here is a comparison of slate 40 vs slate 60 so you can see how it looks

SLATE 40

![Screenshot 2023-08-28 at 1 19 40 PM](https://github.com/infor-design/enterprise/assets/814283/2b40caec-66b6-43b7-8f7d-cced8f04ccb0)

SLATE 60
![Screenshot 2023-08-28 at 1 19 01 PM](https://github.com/infor-design/enterprise/assets/814283/9ae4116f-dfd9-4b5d-8568-42d27c2de2ab)

Working through the other points

talkaboutdesign commented:
> @laurensteiger after making the dark mode slate 90 background and slate 40. I still think it looks kinda boxy if thats the best word. Here is a comparison of slate 40 vs slate 60 so you can see how it looks.
> 
> SLATE 60
> 
> ![Screenshot 2023-08-28 at 1 19 40 PM](https://user-images.githubusercontent.com/814283/263774784-2b40caec-66b6-43b7-8f7d-cced8f04ccb0.png)
> 
> SLATE 40 ![Screenshot 2023-08-28 at 1 19 01 PM](https://user-images.githubusercontent.com/814283/263774786-9ae4116f-dfd9-4b5d-8568-42d27c2de2ab.png)
> 
> Maybe its fine.. Working through the other points
> 
> Rest of the points addressed.

@laurensteiger can you double check that slate colors in your Figma were correct. I agree with @tmcconechy that slate 4 feels a little too much. All other updates look really great. Whatever we land on this border color should also be reflected in other elements like input fields.

tmcconechy commented:
@talkaboutdesign @laurensteiger  i think these border colors can be different than input borders since they arent quite the same thing to me. Its more about "inner" borders vs the outside border of things. But welcome any changes on it.

Also these changes do look good. One other thing to make it perfect is the new selection state https://7814-datagrid-design-updates-enterprise.demo.design.infor.com/components/datagrid/example-multiselect.html?theme=new&mode=dark&colors=default I'm observing that its the same as the header color. Maybe we can use an rgba(0,0,0,.3) sort of color or darker color #161618 ? But open to other suggestions

laurensteiger commented:
@tmcconechy @talkaboutdesign looking at the stroke now, yes slate06 looks better than 4. Let's do 6.

laurensteiger commented:
@tmcconechy @talkaboutdesign as for the selected state. Yea i was going to suggest slate10, which is the #161618.

laurensteiger commented:
Also would like to add that if we are changing the inputs to slate06 in dark mode then we would also have to enhance editor again.

tmcconechy commented:
i think they look pretty good in slate 06 (better to me)

![Screenshot 2023-08-29 at 10 43 15 AM](https://github.com/infor-design/enterprise/assets/814283/4411a957-1cd4-49ea-8f65-e87a8060a436)

tmcconechy commented:
Everything else here should be complete. I dont mind if we make changing darkmode inputs a future question

talkaboutdesign commented:
@laurensteiger @tmcconechy lets note that we can make enhancements to Editor in dark mode later to match slate 06.

tmcconechy commented:
@jbrcna can you give this one a quick approval/look? Would like to merge ASAP

glenlieorillo reviewed (changes requested):
- Text searchfield looks off.

Steps to reproduce:
1. Click on ellipses above table
2. Click 'Personalize Column' from the dropdown
3. Notice the text searchfield

<img width="1792" alt="image" src="https://github.com/infor-design/enterprise/assets/102151352/c49ab0df-bacd-4754-8e47-5afd5854f60a">

<img width="1792" alt="image" src="https://github.com/infor-design/enterprise/assets/102151352/de0f1416-f5a9-481b-9303-f166938659d0">


<img width="1792" alt="image" src="https://github.com/infor-design/enterprise/assets/102151352/6147614f-88cf-42d3-af92-f0bcac922a71">

<img width="1792" alt="image" src="https://github.com/infor-design/enterprise/assets/102151352/160b92f7-5a8f-4612-91cd-cda46fe44ff1">

<img width="1792" alt="image" src="https://github.com/infor-design/enterprise/assets/102151352/a0bfbab6-145c-4b6b-826c-f49503463913">


- Also, close button should be aligned properly.

Steps to reproduce:
1. Click on ellipses above table
2. Click 'Personalize Column' from the dropdown
3. Type in any text in the searchfield
4. Notice the close button

<img width="1792" alt="image" src="https://github.com/infor-design/enterprise/assets/102151352/2e399437-ccd7-4e3d-beb7-df252c96ad16">

jbrcna commented:
approving just a few minor observations

https://7814-datagrid-design-updates-enterprise.demo.design.infor.com/components/datagrid/example-index.html
<img width="1298" alt="image" src="https://github.com/infor-design/enterprise/assets/56722184/652e6cc3-f5ea-4b3f-8eb6-cdeb3647d85f">

<img width="1261" alt="image" src="https://github.com/infor-design/enterprise/assets/56722184/5062efc8-94a9-4d07-91f4-34e399307b41">

https://7814-datagrid-design-updates-enterprise.demo.design.infor.com/components/datagrid/example-multiselect
<img width="1313" alt="image" src="https://github.com/infor-design/enterprise/assets/56722184/f0e423ce-35e7-4281-96aa-04496ad551de">

https://7814-datagrid-design-updates-enterprise.demo.design.infor.com/components/datagrid/example-grouped-headers.html
<img width="1304" alt="image" src="https://github.com/infor-design/enterprise/assets/56722184/0df5adf9-71f2-4ebf-9690-c6781d16b240">

https://7814-datagrid-design-updates-enterprise.demo.design.infor.com/components/datagrid/example-grouped-headers.html
<img width="1313" alt="image" src="https://github.com/infor-design/enterprise/assets/56722184/2456cc35-ff96-44fe-a8ea-cf1a9a1305db">

https://7814-datagrid-design-updates-enterprise.demo.design.infor.com/components/datagrid/example-nested-grids.html
<img width="1161" alt="image" src="https://github.com/infor-design/enterprise/assets/56722184/a598d952-bb98-4656-8f40-12370d7bcc73">


**Question**
where is https://7814-datagrid-design-updates-enterprise.demo.design.infor.com/components/datagrid/example-header-background-color.html?theme=classic&mode=light&colors=default
<img width="1287" alt="image" src="https://github.com/infor-design/enterprise/assets/56722184/a2665be1-72a6-47fc-b0cb-27f064d0d7b1">

jbrcna commented:
**Dark mode** 
<img width="1563" alt="image" src="https://github.com/infor-design/enterprise/assets/56722184/20b35726-87db-4c52-8315-ec876da846f4">

<img width="1364" alt="image" src="https://github.com/infor-design/enterprise/assets/56722184/436a7f8f-b7d5-4456-91db-16359b787512">

talkaboutdesign commented:
> **Dark mode** <img alt="image" width="1563" src="https://user-images.githubusercontent.com/56722184/264625789-20b35726-87db-4c52-8315-ec876da846f4.png">
> 
> #### 
> <img alt="image" width="351" src="https://user-images.githubusercontent.com/56722184/264626236-f3324f4c-b877-4c6c-aa0c-2698e9b6f008.png">
> <img alt="image" width="1364" src="https://user-images.githubusercontent.com/56722184/264625392-436a7f8f-b7d5-4456-91db-16359b787512.png">
> #### 
> <img alt="image" width="309" src="https://user-images.githubusercontent.com/56722184/264626647-3edfca39-4dc2-4d8e-8e87-9b363ec88193.png">

@laurensteiger can you double check this. They are not supposed to be white. @jbrcna they are azure by design. Lauren please double check is they are the right azure.

jbrcna commented:
thanks for clarifying @talkaboutdesign

tmcconechy commented:
@jbrcna @glenlieorillo @talkaboutdesign @laurensteiger 

1. I fixed the wierd personalization dialog
2. the setting for https://7814-datagrid-design-updates-enterprise.demo.design.infor.com/components/datagrid/example-header-background-color.html?theme=classic&mode=light&colors=default is now removed. We dont need it because the header is now grey anyways so no need for this option.
3. I kept the drag state as is because i dont have too much control. I dont think it should be white
4. We decided to not change link color to white after all so the design is incorrect. We kept the link color as it always was.

So i think this is all good

laurensteiger commented:
@tmcconechy the link is giving me a 404? Unless i am doing something wrong.

tmcconechy commented:
@laurensteiger yeah the link is gone. I removed it. we dont need that example anymore (it was the one with the white header). So just ignore.

jbrcna reviewed (commented):
@tmcconechy , this is now working, however, the x icon is not visible in mobile view
 iOS
<img width="188" alt="image" src="https://github.com/infor-design/enterprise/assets/56722184/814d7a79-eb1e-40da-868f-d348edb1e253">

Android
<img width="198" alt="image" src="https://github.com/infor-design/enterprise/assets/56722184/86b7683c-834b-4afe-8e0f-2ff8fb2e549a">

Devtools
<img width="323" alt="image" src="https://github.com/infor-design/enterprise/assets/56722184/492768e7-7b64-4050-a169-10a279ecb4be">

Also, observe the line in searchfield when hovered.
<img width="793" alt="image" src="https://github.com/infor-design/enterprise/assets/56722184/825d9d23-e346-479e-8b7d-067f17edd7a5">

tmcconechy commented:
@jbrcna fixed this issue. I will soon merge this we can take any thing else as a new issue

jbrcna commented:
fix confirmed. thanks @tmcconechy 
<img width="442" alt="image" src="https://github.com/infor-design/enterprise/assets/56722184/d310d12b-f185-4f1b-945a-faee8f750fd8">

<img width="477" alt="image" src="https://github.com/infor-design/enterprise/assets/56722184/70890d06-9d42-4a4d-8f8a-60cd00a32047">

<img width="551" alt="image" src="https://github.com/infor-design/enterprise/assets/56722184/982250a1-c34f-45bb-b956-f20f3648e41b">
<img width="1354" alt="image" src="https://github.com/infor-design/enterprise/assets/56722184/667e5fce-aa87-41e0-b322-5ce4e8fc0161">
